tracing.go

```diff
@@ -347,6 +347,44 @@ func (s *Span) SetDynamicSamplingContext(dsc DynamicSamplingContext) {
 	}
 }
 
+// shouldIgnoreStatusCode checks if the transaction should be ignored based on HTTP status code.
+func (s *Span) shouldIgnoreStatusCode(hub *Hub) bool {
+	if !s.IsTransaction() {
+		return false
+	}
+
+	client := hub.Client()
+	if client == nil {
+		return false
+	}
+
+	ignoreStatusCodes := client.options.TraceIgnoreStatusCodes
+	if len(ignoreStatusCodes) == 0 {
+		return false
+	}
+
+	s.mu.Lock()
+	statusCodeData, exists := s.Data["http.response.status_code"]
```

Comment:
I don't think it's a good idea to implement it like this. I would rather check for the status code in our integrations and drop the transaction there.